py4DSTEM/process/latticevectors/index.py

```diff
@@ -53,3 +53,158 @@ def index_bragg_directions(x0, y0, ux, uy, vx, vy, bragg_x, bragg_y):
+    """
+    Returns a full reciprocal lattice stretching to the limits of the diffraction pattern
+    by making linear combinations of the lattice vectors up to (±h_max,±k_max).
+
+    This can be useful when (for whatever reason, such as polycrystallininty or weird intensities)
+    cause false peaks or missing peaks in the braggvectormap, which can cause errors in the 
+    strain finding routines that rely on those peaks for indexing. This allows us to create
+    a reference lattice that has all combinations of the lattice vectors all the way out to the 
+    edges of the frame, and excluding any erroneous intermediate peaks. 
+
+    Accepts:
+        ux, uy, vx, vy          (float) x and y coords of the u,v lattice vectors
+        x0, y0                  (float) x,y origin of the lattice
+        Q_Nx, Q_Ny              (int) diffraction pattern size (i.e. dc.Q_Nx, dc.Q_Ny)
+        h_max, k_max            (int) maximal indices for generating the lattice
+                                    (the lattive is always trimmed to fit inside the 
+                                     pattern so you can overestimate these, or leave
+                                     unspecified and they will be automatically found)
+
+    Returns:
+        ideal_lattice           (PointList) A 4-coordinate PointList, ('qx','qy','h','k'),
+                                    containing points corresponding to linear combinations
+                                    of the u and v vectors, with associated indices
+    """
+
+    # Matrix of lattice vectors
+    beta = np.array([[ux,uy],[vx,vy]])
+
+    # If no max index is specified, (over)estimate based on image size
+    if (h_max is None) or (k_max is None):
+        (y,x) = np.mgrid[0:Q_Ny,0:Q_Nx]
+        y -= y0
+        x -= x0
+        h_max = np.max(np.ceil(np.abs((x/ux,y/uy))))
+        k_max = np.max(np.ceil(np.abs((x/vx,y/vy))))
+    
+    (hlist,klist) = np.meshgrid(np.arange(-h_max,h_max+1),np.arange(-k_max,k_max+1))
+
+    M_ideal = np.vstack((hlist.ravel(),klist.ravel())).T
+    ideal_peaks = np.matmul(M_ideal,beta)
+
+    ideal_data = np.core.records.fromarrays(np.array([ideal_peaks[:,0],ideal_peaks[:,1],M_ideal[:,0],M_ideal[:,1]]),\
+                                        names='qx,qy,h,k',formats='float,float,int,int')
+
+    coords = [('qx',float),('qy',float),('h',int),('k',int)]
+    ideal_lattice = PointList(coordinates=coords)
+    ideal_lattice.add_dataarray(ideal_data)
+
+    #shift to the DP center
+    ideal_lattice.data['qx'] += x0
+    ideal_lattice.data['qy'] += y0
+
+    # trim peaks outside the image
+    deletePeaks = (ideal_lattice.data['qx'] > Q_Nx) | \
+            (ideal_lattice.data['qx'] < 0) | \
+            (ideal_lattice.data['qy'] > Q_Ny) | \
+            (ideal_lattice.data['qy'] < 0)
+    ideal_lattice.remove_points(deletePeaks)
+
+    return ideal_lattice
+
+def index_point_list_array(pointlistarray, lattice, maxPeakSpacing, mask=None):
```

Comment:
Recommend changing to 

def index_pointlistarray(...

for consistency.

Reply:
fixed
(I forgot the convention 😛)